src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -15,23 +15,89 @@
 
 #include "src/core/lib/event_engine/posix_engine/posix_engine.h"
 
+#include <atomic>
+#include <chrono>
 #include <string>
 #include <utility>
 
 #include "absl/container/flat_hash_set.h"
 #include "absl/functional/any_invocable.h"
+#include "absl/status/status.h"
 
 #include <grpc/event_engine/event_engine.h>
+#include <grpc/event_engine/memory_allocator.h>
 #include <grpc/support/log.h>
 
 #include "src/core/lib/debug/trace.h"
+#include "src/core/lib/event_engine/poller.h"
 #include "src/core/lib/event_engine/posix_engine/timer.h"
 #include "src/core/lib/event_engine/trace.h"
 #include "src/core/lib/event_engine/utils.h"
+#include "src/core/lib/gprpp/sync.h"
+
+#ifdef GRPC_POSIX_SOCKET_TCP
+#include "src/core/lib/event_engine/posix_engine/event_poller_posix_default.h"
+#endif  // GRPC_POSIX_SOCKET_TCP
+
+using namespace std::chrono_literals;
 
 namespace grpc_event_engine {
 namespace experimental {
 
+#ifdef GRPC_POSIX_SOCKET_TCP
+using grpc_event_engine::posix_engine::PosixEventPoller;
+
+PosixEventEngine::PosixEventEngine(PosixEventPoller* poller)
+    : poller_(poller), poller_state_(PollerState::kExternal) {
+  GPR_ASSERT(poller_ != nullptr);
+}
+
+PosixEventEngine::PosixEventEngine()
+    : poller_(grpc_event_engine::posix_engine::GetDefaultPoller(this)) {
+  ++shutdown_ref_;
+  if (poller_ != nullptr) {
+    executor_.Run([this]() { PollerWorkInternal(); });
+  }
+}
+
+void PosixEventEngine::PollerWorkInternal() {
+  // TODO(vigneshbabu): The timeout specified here is arbitrary. For instance,
+  // this can be improved by setting the timeout to the next expiring timer.
+  auto result = poller_->Work(24h, [this]() {
+    ++shutdown_ref_;
+    executor_.Run([this]() { PollerWorkInternal(); });
+  });
+  if (result == Poller::WorkResult::kDeadlineExceeded) {
+    // The event engine is not shutting down but the next asynchronous
+    // PollerWorkInternal did not get scheduled. Schedule it now.
+    ++shutdown_ref_;
```

Comment:
shutdown_ref_.fetch_add(1)

Reply:
Removed the implementation. Now we dont need a shutdown_ref. Instead the poller state is managed by a PosixEnginePollerManager class which is held as a shared_ptr by the event engine. In the event engine destructor, we no simple trigger shutdown on this class.